parsl/executors/high_throughput/executor.py

```diff
@@ -455,9 +458,11 @@ def _result_queue_worker(self):
         """
         logger.debug("Result queue worker starting")
 
-        while not self.bad_state_is_set:
+        while not self.bad_state_is_set and not self._result_queue_thread_exit.is_set():
             try:
-                msgs = self.incoming_q.get()
+                msgs = self.incoming_q.get(timeout_ms=self.poll_period)
+                if msgs is None:  # timeout
+                    continue
```

Comment:
Oh good.  This exact procedure has been on my mind for quite awhile, so thank you for this!

I'm thinking at some point we can go farther and simply block completely, but I haven't fully gotten through the shutdown logic for this route yet.  But "on my mind," at some point.